d3a_interface/sim_results/kpi.py

```diff
@@ -145,11 +145,74 @@ def update_area_kpi(self, area_dict, core_stats):
         self._accumulate_energy_trace(core_stats)
 
 
+class SavingsKPI:
+    def __init__(self):
+        self.producer_ess_set = set()
+        self.consumer_ess_set = set()
+        self.house_list = list()
+        self.fit_revenue = 0.
+        self.utility_bill = 0.
+        self.base_case_revenue = 0.
+        self.d3a_revenue = 0.
+        self.saving_absolute = 0.
+        self.saving_percentage = 0.
+
+    def calculate_saving_kpi(self, area_dict, core_stats, gf_alp):
```

Comment:
```suggestion
    def calculate_savings_kpi(self, area_dict, core_stats, gf_alp):
```

Reply:
This method is the main feature of this PR but it's not being unit-tested as far as I see. I suggest adding tests for it and for the individual smaller methods (e.g. `get_feed_in_tariff_rate_excluding_path_grid_fees` and the others).

Reply:
`savings_kpi` with grid fees being tested via integration_tests, so not sure if still the need arise to write unittests